app/server/lib/SamlConfig.ts

```diff
@@ -204,6 +198,43 @@ export class SamlConfig {
     }));
   }
 
+  private async _processInitialRequest(req: express.Request) {
+    const relayState: string = req.body.RelayState;
+    const sessionId = this._gristServer.getSessions().getSessionIdFromRequest(req) || undefined;
+    const params = {
+      sessionId,
+      redirectUrl: "",
+      unsolicited: true,
+      action: "",
+    };
+
+    if (!relayState) {
+      // Presumably an IdP-inititated signin.
+      params.redirectUrl = checkRedirectUrl(relayState, req).href;
+      params.action = "login";
+      return params;
+    }
+
+    const permitStore = this._gristServer.getExternalPermitStore();
+    const state = await permitStore.getPermit(relayState);
+    if (!state) {
+      // Presumably an IdP-inititated signin without a relayState.
```

Comment:
"without a relayState" seems to refer to the other clause (where `!relayState` is checked). Perhaps comments got mixed up?